### PR TITLE
add receivedAt and response utilities for tool calling

### PR DESCRIPTION
**Pull request by zachkrall**
- add `receivedAt` attribute to `ToolCall` message
- add `response.success` and `response.error` utilities to make sending the correct response easier
- remove async from event listener on client

### CODE REVIEW

**Review by hume-brian — APPROVED** [review]

File: packages/react/src/lib/useVoiceClient.ts [hunk 1]
```diff
@@ -92,14 +105,41 @@ export const useVoiceClient = (props: {
...
+                error,
+                code,
+                level,
+                content,
+              }),
+            })
+            .then((response) => {
+              const parsed = ToolResponseSchema.safeParse(response);
+              if (parsed.success) {
+                client.current?.sendToolResponse(parsed.data);
```
Comment (hume-brian) [inline packages/react/src/lib/useVoiceClient.ts:138]: should make this `sendToolError` if it's an error or remove the `sendToolError` method

Comment (zachkrall) [reply packages/react/src/lib/useVoiceClient.ts:138]: @hume-brian oh, this a block you added in #188.

https://github.com/HumeAI/empathic-voice-api-js/pull/188/files#diff-606d0dd18ceb7429e2c5d6095b5a67db2448400e9ba863976beeaac5cfafc32cR98

@yinishi do you think this should be something like 

```typescript
z.union([ToolResponseSchema, ToolErrorSchema]).safeParse(response)
```

Comment (zachkrall) [reply packages/react/src/lib/useVoiceClient.ts]: @hume-brian @yinishi I updated the zod parsing and function name :)

Comment (yinishi) [reply packages/react/src/lib/useVoiceClient.ts]: lgtm!

**Pull request merged**